pkg/download/dependency_resolution.go

```diff
@@ -132,8 +133,19 @@ func collectConfigsById(configs project.ConfigsPerType) map[string]config.Config
 		for _, conf := range configs {
 			configsById[conf.Coordinate.ConfigId] = conf
 			if conf.OriginObjectId != "" {
+				// resolve Settings references by Object ID as well
 				configsById[conf.OriginObjectId] = conf
 			}
+			if conf.OriginObjectId != "" && conf.Coordinate.Type == "builtin:management-zones" {
+				// resolve Management Zone Settings by Numeric ID as well
+				numID, err := idutils.GetNumericIDForObjectID(conf.OriginObjectId)
+				if err != nil {
+					log.Warn("Failed to decode numeric ID of config %q, auto-resolved references may be incomplete: %v", conf.Coordinate, err)
+				} else {
+					strId := fmt.Sprintf("%d", numID)
+					configsById[strId] = conf
+				}
+			}
```

Comment:
This part seems to be not covered by any tests.
If it's easy to provide a test case for it pls do it, otherwise I guess it's fine

Reply:
Added some tests, but didn't find a reasonable way to test two of the error cases - they should never happen if the previous checks have passed

Reply:
I was more like thinking of adding specific test cases to `dependency_resolution_test.go`

Reply:
ah... right, apparently reading is not my strong suite, missed on what you had actually commented 😅 

Yes, I will use the known IDs used in the idutils test cases for some dependency cases as well 👍

Reply:
test extended & commits cleaned up